bin/match_filenames.py

```diff
@@ -1 +1,155 @@
+                filenames = [l for l in f.read().split("\n") if l]
+
+        equivalence_classes = classify_list(filenames, pattern)
+
+        all_classes[path] = equivalence_classes
+        all_files[path] = filenames
+
+    return all_classes, all_files
+
+
+def classify_list(filenames, pattern):
+    # sort each filename into a class
+    # and generate a mapping from class to filenames belonging to that class
+    equivalence_classes = defaultdict(list)
+    for filename in filenames:
+        # todo: add nicer error message when the regexp is invalid
+        match = re.search(pattern, filename)
+        if not match:
+            class_ = filename
+        elif len(match.groups()) == 0:
+            class_ = match.group(0)
+        else:
+            class_ = "".join(match.groups())
+
+        equivalence_classes[class_].append(filename)
+    return equivalence_classes
+
+
+def flatten(list_of_lists):
+    return it.chain.from_iterable(list_of_lists)
+
+
+def move_files(path, diff_files, all_files, suffix):
+    target = path + suffix
+    print("Moving {}/{} files from '{}' to '{}'".format(len(diff_files), len(all_files), path, target))
+
+    if os.path.isdir(path):
+        for file in diff_files:
+            os.makedirs(target, exist_ok=True)
+            shutil.move(join(path, file), target)
+    else:
+        with open(target, "w") as f:
+            f.write("\n".join(sorted(diff_files)))
+        with open(path, "w") as f:
+            remaining = set(all_files) - set(diff_files)
+            f.write("\n".join(sorted(remaining)))
+
+
+def delete_files(path, diff_files, all_files):
+    print("Deleting {}/{} files in '{}'".format(len(diff_files), len(all_files), path))
+
+    if os.path.isdir(path):
+        for file in diff_files:
+            os.remove(join(path, file))
+    else:
+        with open(path, "w") as f:
+            remaining = set(all_files) - set(diff_files)
+            f.write("\n".join(sorted(remaining)))
+
+
+def print_list(lines, verbose, limit=20):
+    if not lines:
+        return
+    if verbose or len(lines) <= limit:
+        print("\n".join(list(lines)))
+    else:
+        print("\n".join(list(lines)[:limit]))
+        print("...")
+
+
+@click.command()
+@click.option("-p", "--path_and_pattern", type=click.Tuple([click.Path(exists=True), str]), multiple=True, required=True,
+              help="(path, pattern) tuple. path can be a directory or a text file. pattern is a regular "
+                   "expression whose matches define equivalence classes. If there are subgroups, use those. If there "
+                   "is no match, use the whole filename to define a class.")
+@click.option("-t", "--target", type=click.Choice(["diff", "int"]), default="diff",
+              help="whether to do move/delete on the difference or the intersection (default: diff)")
+@click.option("--delete", is_flag=True,
+              help="whether to delete the difference or intersection")
```

Comment:
ohhhh

Reply:
vielleicht oben: whether to calculate the difference or the intersection

Reply:
und dann hier 
```suggestion
              help="whether to delete the resulting files")
```
oder so in der Richtung